### PR TITLE
(FACT-3184) Prefer require_relative

### PR DESCRIPTION
**Pull request by joshcooper**
Use `require_relative` instead of `require` to reduce amount of file IO due to scanning the `$LOAD_PATH`.

Don't do `Dir.glob` to require facts and resolvers, because the order is not stable prior to Ruby 3. Also explicitly requiring files makes it clear how files are being required and in what order. For example, there were several circular requires (file_loader.rb required itself) and we were Dir.globbing non-existent directories.

Passed integration tests against all supported platforms:

- [x] [puppet-agent#7.x](https://jenkins-platform.delivery.puppetlabs.net/view/puppet-agent/view/PR%20testing/job/platform_puppet-agent-extra_puppet-agent-integration-suite_pr/2424/)
- [x] [puppet-agent#main](https://jenkins-platform.delivery.puppetlabs.net/view/puppet-agent/view/PR%20testing/job/platform_puppet-agent-extra_puppet-agent-integration-suite_pr/2422/)

### CODE REVIEW

File: lib/facter/framework/core/file_loader.rb [hunk 1]
```diff
@@ -15,7 +15,10 @@ def load_dir(*dirs)
   return unless Dir.exist?(folder_path.tr('*', ''))
 
   files_to_require = Dir.glob(File.join(folder_path, '*.rb')).reject { |file| file =~ %r{/ffi/} }
-  files_to_require.each(&method(:require))
+  files_to_require.each do |file|
+    rpath = file[File.dirname(__FILE__).length+1..-1]
+    require_relative rpath
```
Comment (joshcooper) [inline lib/facter/framework/core/file_loader.rb:20]: Probably a better way to do this using pathname's `relative_path_from` or at least compute the prefix length once.

**Comment by joshcooper** [comment]
jenkins please test this on all

**Comment by joshcooper** [comment]
jenkins please test this on all

**Comment by joshcooper** [comment]
jenkins please test this on aix71-POWERfa,aix72-POWERfa,amazon6-64a,amazon7-64a,debian10-64a,debian11-64a,fedora36-64a,osx1015-64a,osx11-64a,osx12-64a,osx12-ARM64a,redhat6-64a,redhat6-32a,redhat7-64a,redhat7-AARCH64a,redhat8-64a,redhat8-POWERa,redhat8-AARCH64a,redhat9-64a,redhatfips7-64a,redhatfips8-64a,almalinux8-64a,rocky8-64a,sles12-64a,sles15-64a,solaris10-64a,solaris10-SPARCa,solaris114-64a,solaris11-SPARCa,ubuntu1804-64a,ubuntu1804-AARCH64a,ubuntu2004-64a,ubuntu2004-AARCH64a,ubuntu2204-64a,windows10ent-64a,windows10ent-32a,windows11ent-64a,windows2012r2-64a,windows2019_ja-64a%7Blocale=ja%7D,windows2016-64a,windows2019-64a,windows2022-64a,windowsfips2012r2-64a with agent 7.x

**Review by mhashizume — APPROVED** [review]

**Pull request merged**